src/test/java/seedu/address/logic/commands/ListTaskCommandTest.java

```diff
@@ -23,14 +27,37 @@ public class ListTaskCommandTest {
     @BeforeEach
     public void setUp() {
         model = new ModelManager(getTypicalAddressBook(), getTypicalTaskList(), new UserPrefs());
-        new ListTaskCommand(new DefaultComparator()).execute(model);
         expectedModel = new ModelManager(model.getAddressBook(), model.getTaskList(), new UserPrefs());
     }
 
     @Test
-    public void execute_listIsNotFiltered_showsSameList() {
+    public void execute_listIsNotFilteredDefaultComparator_showsSameList() {
         assertCommandSuccess(new ListTaskCommand(new DefaultComparator()),
-                model, ListTaskCommand.MESSAGE_SUCCESS, expectedModel);
+                model, ListTaskCommand.MESSAGE_LIST_DEFAULT_SUCCESS, expectedModel);
+    }
+
+    @Test
+    public void execute_listIsFilteredDefaultComparator_showsEverything() {
+        showTaskAtIndex(model, INDEX_FIRST);
+        Assertions.assertTrue(model.getTaskList().equals(expectedModel.getTaskList()));
+        assertCommandSuccess(new ListTaskCommand(new DefaultComparator()),
+                model, ListTaskCommand.MESSAGE_LIST_DEFAULT_SUCCESS, expectedModel);
+    }
+
+    @Test
+    public void execute_listIsNotFilteredDeadlineComparator_showsSameList() {
+        expectedModel.updateSortedTaskList(new DeadlineComparator());
```

Comment:
Should a new expectedModel local variable be defined here to avoid unexpected mutations?

Reply:
Not a bug as the model is reset in a beforeEach